.github/workflows/ubuntubuilds.yml

```diff
@@ -89,10 +120,19 @@ jobs:
           submodules: recursive
 
       - name: Build
-        run:
-          mkdir build && cd build && cmake -DKPSR_WITH_DOXYGEN=true -DKPSR_WITH_DDS=false -DKPSR_WITH_ZMQ=true \
-          -DKPSR_TEST_PERFORMANCE=false -DKPSR_WITH_SOCKET=true -DKPSR_WITH_YAML=true -DKPSR_WITH_CODE_METRICS=true -DCMAKE_PREFIX_PATH=/opt/klepsydra/thirdparties \
-          -DCMAKE_BUILD_TYPE=Debug .. && make -j ${nproc}
+        run: |
+          mkdir build && cd build 
+          cmake -DKPSR_WITH_DOXYGEN=true -DKPSR_WITH_DDS=false -DKPSR_WITH_ZMQ=true -DKPSR_TEST_PERFORMANCE=false -DKPSR_WITH_SOCKET=true -DKPSR_WITH_YAML=true -DKPSR_WITH_CODE_METRICS=true -DCMAKE_PREFIX_PATH=/opt/klepsydra/thirdparties -DCMAKE_BUILD_TYPE=Debug ../
+          make -j ${nproc}
 
       - name: Test
-        run: cd build && make test
+        run: cd build && make test ARGS="--gtest_shuffle"
```

Comment:
Enable code coverage here instead so that it tests all the modules. Otherwise zmq and yaml code is not checked for coverage.